src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -1,47 +1,459 @@
+			volumes = response.volumes();
+
+		}
+		return volumes;
+	}
+
+	protected DescribeVolumesResponse getDescribeVolume(String volumeId, Ec2Client client) {
+		DescribeVolumesRequest request = DescribeVolumesRequest.builder()
+				.volumeIds(volumeId)
+				.build();
+
+		return client.describeVolumes(request);
+	}
+
+	protected List<String> getVolumeIds(Instance instance) {
+		List<String> volumeIds = new ArrayList<String>();
+		for (int i = 0; i < instance.blockDeviceMappings().size(); i++) {
+			volumeIds.add(instance.blockDeviceMappings().get(i).ebs().volumeId());
+		}
+		return volumeIds;
+	}
+
+	protected Instance getInstanceReservation(DescribeInstancesResponse response) throws InstanceNotFoundException {
+		if (!response.reservations().isEmpty()) {
+			Reservation reservation = response.reservations().get(FIRST_POSITION);
+			if (!reservation.instances().isEmpty()) {
+				return reservation.instances().get(FIRST_POSITION);
+			}
+		}
+		throw new InstanceNotFoundException();
+	}
+	
+	protected List<InstanceNetworkInterfaceSpecification> loadNetworkInterfaces(ComputeOrder computeOrder) {
+		List<String> networkIds = getNetworkIds(computeOrder);
+		List<InstanceNetworkInterfaceSpecification> networkInterfaces = new ArrayList<>();
+		InstanceNetworkInterfaceSpecification nis = null;
+		for (String networkId : networkIds) {
+			nis = createInstanceNetworkInterface(networkId);
+			networkInterfaces.add(nis);
+		}
+		return networkInterfaces;
+	}
 
-    @Override
-    public boolean hasFailed(String instanceState) {
-        return true;
-    }
+	protected List<String> getNetworkIds(ComputeOrder computeOrder) {
+		List<String> networkIds = new ArrayList<String>();
+		networkIds.add(this.subnetId);
+		if (computeOrder.getNetworkIds() != null) {
+			networkIds.addAll(computeOrder.getNetworkIds());
+		}
+		return networkIds;
+	}
+
+	protected InstanceNetworkInterfaceSpecification createInstanceNetworkInterface(String networkId) {
+		InstanceNetworkInterfaceSpecification networkInterface = InstanceNetworkInterfaceSpecification.builder()
+				.subnetId(networkId) // Default sub-net in the available zone of the selected region.
+				.deviceIndex(FIRST_POSITION) // The first position of the network interface.
+				.groups(this.securityGroup).build();
+		
+		return networkInterface;
+	}
+
+	protected CreateTagsRequest createInstanceTagName(ComputeOrder computeOrder, String instanceId) {
+		String name = defineInstanceName(computeOrder.getName());
+
+		Tag tagName = Tag.builder()
+				.key(AWS_TAG_NAME)
+				.value(name)
+				.build();
+
+		CreateTagsRequest tagRequest = CreateTagsRequest.builder()
+				.resources(instanceId)
+				.tags(tagName)
+				.build();
+
+		return tagRequest;
+	}
+
+	protected String defineInstanceName(String instanceName) {
+		return instanceName == null ? SystemConstants.FOGBOW_INSTANCE_NAME_PREFIX + getRandomUUID() : instanceName;
+	}
+
+	// This method is used to aid in the tests
```

Comment:
I think not only in tests for this plugin..

Reply:
Okay, I'll remove the comment.